### PR TITLE
Updated pickers to reflect web restrictions

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Revert broken PR https://github.com/expo/expo/pull/11847
- Updated docs to reflect web restrictions regarding the cancel event. I attempted to implement the feature but it's just not possible at the moment. For anyone curious, [here is the branch](https://github.com/expo/expo/tree/%40evanbacon/improve-web-pickers). Differences between Desktop/Mobile Chrome, and Desktop/Mobile Safari have made this too unreliable to even attempt to implement.

# Test Plan

- Updated docs and added basic SSR guards
- Ran NCL on web for desktop and mobile.

### CODE REVIEW

File: apps/native-component-list/src/screens/DocumentPickerScreen.tsx [hunk 1]
```diff
@@ -10,14 +10,21 @@ export default function DocumentPickerScreen() {
   const [document, setDocument] = React.useState<DocumentPicker.DocumentResult | null>(null);
 
   const openPicker = async () => {
+    const time = Date.now();
     const result = await DocumentPicker.getDocumentAsync({
       copyToCacheDirectory: copyToCache,
     });
+    console.log(`Duration: ${Date.now() - time}ms`);
+    console.log(`Results:`, result);
```
Comment (bbarthec) [inline apps/native-component-list/src/screens/DocumentPickerScreen.tsx:18]: Do we want to leak such `console.log`s to production apps? 🤔 
Maybe we can hide them behind some `DEBUG=1` env variable or so? 🤔

**Review by bbarthec — APPROVED** [review]
As for me, it looks good 👍

**Comment by bbarthec** [comment]
@EvanBacon, thank you for spotting and fixing the regression 🎉

File: apps/native-component-list/src/screens/DocumentPickerScreen.tsx (hunk 1 above)
Comment (brentvatne) [reply apps/native-component-list/src/screens/DocumentPickerScreen.tsx:18]: shouldn't matter since it's just in native-component-list

**Review by brentvatne — APPROVED** [review]
let's cherry-pick this to the sdk-41 branch also

**Pull request merged**